feat: update docs to sdk connectors v2 (#984)
FuelLabs/fuels-wallet: packages/docs/docs/dev/reference.mdx

> ## Methods

Reviewer: ```suggestion
```

Should remove and bump everything down a tag to match the hooks reference which looks a lot cleaner

<img width="1200" alt="Screenshot 2023-11-22 at 5 15 15 PM" src="https://github.com/FuelLabs/fuels-wallet/assets/106365423/a616e12d-0f3f-48c0-a762-b22405f4ba5f">

> ## Methods

Reviewer: Can we get the methods here in alphabetical order please?

> ### Listing connectors

Reviewer: ```suggestion
## Listing connectors
```

`connectors(): Promise<Array<FuelConnector>>`<br />
Return connectors with availability metadata.

> ### Has connector

Reviewer: ```suggestion
## Has connector
```

`hasConnector(connectorName?: string | FuelConnector): Promise<boolean>`<br />
Check if any Connector is avialable or if a connector is provided check if it is available.

> ### Current connector

Reviewer: ```suggestion
## Current connector
```

`currentConnector(): FuelConnector`<br />
Return the current selected connector.

> ### Get connector

Reviewer: ```suggestion
## Get connector
```

`getConnector(connectorName: string): FuelConnector`<br />
Return the connector with the given name.

> ### Selecting connector

Reviewer: ```suggestion
## Selecting connector
```

`selectConnector(connectorName: string): Promise<boolean>` <br />
This method will check if the desired connector is installed. If it is not detected in 1 second, the method throws an error.

> ### Version

Reviewer: ```suggestion
## Version
```

`version(): Promise<string>`<br />
Return the current version of the Wallet and Network supported.

> ### Connect

Reviewer: ```suggestion
## Connect
```

`connect(): Promise<boolean>`<br />
Request permission to start a connection between the project and the wallet. Return the connection state.

> ### Disconnect

Reviewer: ```suggestion
## Disconnect
```

`disconnect(): Promise<boolean>`<br />
Disconnect your project from the wallet. Return the connection state.

> ### Get connection status

Reviewer: ```suggestion
## Get connection status
```

`isConnected(): Promise<boolean>`<br />
Return the state of the application connection.

> ### List Accounts

Reviewer: ```suggestion
## List accounts
```

`accounts(): Promise<Array<string>>`<br />
Return the accounts authorized for the current connection.

> ### Current Account

Reviewer: ```suggestion
## Current account
```

`currentAccount(): Promise<string | null>`<br />
Return the current account being used in the wallet application.
If the account selected is not authorized for the connection, returns null.

> ### List Networks

Reviewer: ```suggestion
## List networks
```

`networks(): Promise<Array<Network>>`<br />
Return the list of networks in the current wallet.

> ### Current Network

Reviewer: ```suggestion
## Current network
```

`currentNetwork(): Promise<Network>`<br />
Return the current network being used in the wallet application.

> ### Select Network

Reviewer: ```suggestion
## Select network
```

`selectNetwork(url: string): Promise<boolean>`<br />
Request the user to switch to a different network. Return true if success or false it fails.

> ### Add Network

Reviewer: ```suggestion
## Add network
```

`addNetwork(url: string): Promise<boolean>`<br />
Request the user to add a new network. Return true if success or false it fails.

> ### List Assets

Reviewer: ```suggestion
## List assets
```

`assets(): Promise<Array<Asset>>`<br />
Return the list of assets in the current wallet.

> ### Add Asset

Reviewer: ```suggestion
## Add asset
```

`addAsset(asset: Asset): Promise<boolean>`<br />

> Request the user to add a asset metadata to the wallet.

Reviewer: ```suggestion
Request the user to add metadata for an asset to the wallet.
```

> ### Add Assets

Reviewer: ```suggestion
## Add assets
```

`addAssets(asset: Asset[]): Promise<boolean>`<br />

> Request the user to add multiple assets metadata at once to the wallet.

Reviewer: ```suggestion
Request the user to add metadata for multiple assets to the wallet at once.
```

> ### Get ABI

Reviewer: ```suggestion
## Get ABI
```

`getABI(contractId: string): Promise<JsonAbi>`<br />

> Return ABI of contractId given or null if not found.

Reviewer: ```suggestion
Return the ABI of the given contractId, or null if not found.
```

> ### Has ABI

Reviewer: ```suggestion
## Has ABI
```

`hasABI(contractId: string): Promise<boolean>`<br />

> Return boolean if the ABI for contractId given is found.

Reviewer: ```suggestion
Return a boolean indicating if the ABI for the given contractId is found.
```

> ### Add ABI

Reviewer: ```suggestion
## Add ABI
```

`addAbi(abiMap: AbiMap): Promise<boolean>`<br />

> Add ABI to the users Wallet.

Reviewer: ```suggestion
Add the ABI to the user's wallet.
```

> ### Request signature message

Reviewer: ```suggestion
## Request signature message
```

`async signMessage(address: string, message: string): Promise<string>`<br />
Request a message signature for one specific account

> ### Send transaction

Reviewer: ```suggestion
## Send transaction
```

`async sendTransaction(address: string, transaction: TransactionRequestLike): Promise<string>`<br />

> Request the user specific account to send a transaction.

Reviewer: ```suggestion
Request a specific user account to send a transaction.
```

> ### Get Wallet

Reviewer: ```suggestion
## Get Wallet
```

`getWallet(address: string | AbstractAddress, provider?: FuelWalletProvider): Promise<FuelWalletLocked>`<br />

> Return a `FuelWalletLocked` instance, witch can be used for contract, transfers and other interactions.

Reviewer: ```suggestion
Return a `FuelWalletLocked` instance, which can be used for contracts, transfers, and other interactions.
```

> ### Unsubscribe all

Reviewer: ```suggestion
## Unsubscribe all
```

`unsubscribe(): void`<br />

> Remove all open listeners this is useful when you want to remove the Fuel instance and avoid memory leaks.

Reviewer: ```suggestion
Remove all open listeners. This is useful when you want to dispose of the Fuel instance and avoid memory leaks.
```

> ### Destory

Reviewer: ```suggestion
## Clean
```

Wrong title

`clean(): void`<br />

> Clean the storage from Fuel SDK. Used to keep track of selected connector.

Reviewer: ```suggestion
Clean the storage used by the Fuel SDK, which is utilized to keep track of the selected connector.
```

> ### Destory

Reviewer: ```suggestion
## Destory
```

`destroy(): void`<br />

> Removes all listeners and clean the storage.

Reviewer: ```suggestion
Removes all listeners and cleans the storage.
```

The `fuel` object has an `events` property which is an enum of all the events that can be listened to. <br />
The `on` method takes two arguments, the event name and a callback function. The callback function receives data associated with the event.

```ts
fuel.on(fuel.events.connection, (connectionState) => {
  console.log(connectionState);
});
```

> ### Events

Reviewer: Is there a better name for this? "Events" under "Events" doesnt really make sense

Reviewer: Used "Event Types" for the chart

| Event              | Trigger                   | Params                 |
| ------------------ | ------------------------- | ---------------------- |
| `connectors`       | connectors change         | `Array<FuelConnector>` |
| `currentConnector` | currentConnector change   | `FuelConnector`        |
| `accounts`         | accounts available change | `Array<string>`        |
| `currentAccount`   | current account change    | `string \| null`       |
| `connection`       | connection state change   | `boolean`              |
| `currentNetwork`   | cuurent network change    | `Network`              |

> | `assets`           | assets metadata change    | `Array<Asset>`         |

Reviewer: Chart looks very odd when generated
<img width="1147" alt="Screenshot 2023-11-22 at 5 02 57 PM" src="https://github.com/FuelLabs/fuels-wallet/assets/106365423/31c70fae-a017-475a-b8bd-fcd4b26b7c04">
Should match the docs-hub charts
<img width="755" alt="Screenshot 2023-11-22 at 5 06 28 PM" src="https://github.com/FuelLabs/fuels-wallet/assets/106365423/6ae714d5-e7de-471e-a7ac-6e06cef542fd">
